llvm/lib/IR/ReplaceConstant.cpp

```diff
@@ -85,9 +90,11 @@ bool convertUsersOfConstantsToInstructions(ArrayRef<Constant *> Consts) {
       if (auto *C = dyn_cast<Constant>(U.get())) {
         if (ExpandableUsers.contains(C)) {
           Changed = true;
-          Instruction *NI = expandUser(BI, C);
-          InstructionWorklist.insert(NI);
-          U.set(NI);
+          auto NewInsts = expandUser(BI, C);
+          for (auto *NI : NewInsts)
+            NI->setDebugLoc(BI->getDebugLoc());
```

Comment:
For phi nodes this will take the debug loc from the terminator of the incoming block rather than the phi node -- is that intended?

Reply:
Thank you for the review. I didn't notice this problem. I've updated to use debug loc from the original inst